src/main/java/com/iexec/worker/sgx/SgxService.java

```diff
@@ -40,11 +40,11 @@
     public SgxService(
             WorkerConfigurationService workerConfigService,
             DockerService dockerService,
-            @Value("${debug.forceTeeDisabled}") boolean forceTeeDisabled
+            @Value("${tee.sgx.mode}") SgxMode sgxMode
     ) {
         this.workerConfigService = workerConfigService;
         this.dockerService = dockerService;
-        this.isSgxSupported = !forceTeeDisabled && isSgxSupported();
+        this.isSgxSupported = !(SgxMode.OFF.equals(sgxMode)) && isSgxSupported();
```

Comment:
I tend to prefer the inverted notation with the reference value between parentheses
```suggestion
        this.isSgxSupported = !sgxMode.equals(SgxMode.OFF) && isSgxSupported();
```
Is there a best practice in the team for this ?

Reply:
`SgxMode.OFF.equals(sgxMode)` could look unfamiliar yes, but it freely prevents NPE.

Reply:
I don't think we have any best practice in the team, but I really prefer applying the `equals` method to the constant as it avoids NPEs. I also find it easier to read as you already know what your value should be equal to, before reading the actual var.